src/core/Akka.Persistence.TestKit/PersistenceTestKit.cs

```diff
@@ -0,0 +1,64 @@
+//-----------------------------------------------------------------------
+// <copyright file="PersistenceTestKit.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2019 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2019 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Akka.Persistence.TestKit
+{
+    using Actor;
+    using Akka.TestKit;
+    using Akka.TestKit.Xunit;
+    using Configuration;
+    using System;
+
+    public abstract class PersistenceTestKit : TestKitBase
+    {
+        protected PersistenceTestKit(string actorSystemName = null, string testActorName = null)
+            : base(new XunitAssertions(), GetConfig(), actorSystemName, testActorName)
+        {
+            JournalActorRef = GetJournalRef(Sys);
+            Journal = TestJournal.FromRef(JournalActorRef);
+        }
+
+        public IActorRef JournalActorRef { get; }
+
+        public ITestJournal Journal { get; }
+
+        public void WithFailingJournalRecovery(Action execution)
+        {
+            try
+            {
+                Journal.OnRecovery.Fail();
+                execution();
+            }
+            finally
+            {
+                // restore normal functionality
+                Journal.OnRecovery.Pass();
+            }
+        }
+
+        public void WithFailingJournalWrites(Action execution)
+        {
+            try
+            {
+                Journal.OnWrite.Fail();
```

Comment:
IMHO, we should probably get into the habit of returning `Task` here instead, in order to avoid the types of issues I dsecribe in https://github.com/akkadotnet/akka.net/issues/3854 going forward

Reply:
Likewise, this method should return `Task` too given the previous comment